eva/optimizer/rules/rules.py

```diff
@@ -499,6 +503,64 @@ def apply(self, before: LogicalJoin, context: OptimizerContext):
         yield new_join
 
 
+class ReorderPredicates(Rule):
```

Comment:
Should we give it a more specific name like `CostBasedReorder...` to differentiate with `SelectivityBasedReorder`?